attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{bail, Context};
+use async_trait::async_trait;
+use kbs_types::{Attestation, Challenge, ErrorInformation, Request, Response};
+use log::{debug, warn};
+use resource_uri::ResourceUri;
+use serde::Deserialize;
+use sha2::{Digest, Sha384};
+
+use crate::{
+    api::KbsClientCapabilities,
+    client::{
+        ClientTee, KbsClient, KBS_GET_RESOURCE_MAX_ATTEMPT, KBS_PREFIX, KBS_PROTOCOL_VERSION,
+    },
+    evidence_provider::EvidenceProvider,
+    keypair::TeeKeyPair,
+    token_provider::Token,
+    Error, Result,
+};
+
+#[derive(Deserialize, Debug, Clone)]
+struct AttestationResponseData {
+    // Attestation token in JWT format
+    token: String,
+}
+
+impl KbsClient<Box<dyn EvidenceProvider>> {
+    /// Get a [`TeeKeyPair`] and a [`Token`] that certifies the [`TeeKeyPair`].
+    /// It will check if the client already has a valid token. If so, return
+    /// the token. If not, the client will generate a new key pair and do a new
+    /// RCAR handshaking.
+    pub async fn get_token(&mut self) -> Result<(Token, TeeKeyPair)> {
+        if let Some(token) = &self.token {
+            if token.check_valid().is_err() {
+                self.rcar_handshake()
+                    .await
+                    .map_err(|e| Error::RcarHandshake(e.to_string()))?;
+            }
+        } else {
+            self.rcar_handshake()
+                .await
+                .map_err(|e| Error::RcarHandshake(e.to_string()))?;
+        }
+
+        assert!(self.token.is_some());
+
+        let token = self.token.clone().unwrap();
+        let tee_key = self.tee_key.clone();
+        Ok((token, tee_key))
+    }
+
+    /// Perform RCAR handshake with the given kbs host. If succeeds, the [`KbsBackgoundCheckClient`] will
```

Comment:
I think this is a leftover, the type doesn't exist anymore (and a typo, background is missing an r)

Reply:
Yes. fixed this